src/vmm/src/serial.rs

```diff
@@ -3,31 +3,32 @@
 
 #![cfg(target_arch = "x86_64")]
 
-use std::convert::TryInto;
-use std::io::{stdin, Read, Write};
-
+use crate::console::ConsoleReaderWrapper;
 use event_manager::{EventOps, Events, MutEventSubscriber};
+use std::convert::TryInto;
+use std::io::{Read, Write};
+use utils::debug;
 use vm_device::{
     bus::{PioAddress, PioAddressValue},
     MutDevicePio,
 };
 use vm_superio::Serial;
 use vmm_sys_util::epoll::EventSet;
 
-use utils::debug;
-
 /// Newtype for implementing `event-manager` functionalities.
-pub(crate) struct SerialWrapper<W: Write>(pub Serial<W>);
+pub(crate) struct SerialWrapper<W: Write>(pub Serial<W>, pub ConsoleReaderWrapper);
```

Comment:
This is making the code hard to read. We've been using the newtype pattern only for when we have one type as inner type. Since we are now having 2 types here, we should define it as a struct with named fields.